src/Injector/ConditionalDiscoveryTrait.php

```diff
@@ -4,35 +4,26 @@
 
 namespace Laminas\ComponentInstaller\Injector;
 
+use Laminas\ComponentInstaller\ConfigDiscovery\DiscoveryInterface;
+
 use function str_replace;
 
+/**
+ * @psalm-require-extends AbstractInjector
+ */
 trait ConditionalDiscoveryTrait
```

Comment:
We should make this one `@internal` - lots of foot-gunning with this stuff

Reply:
I marked everything as `@internal` now.